R/rnet_join.R

```diff
@@ -249,16 +250,46 @@ rnet_merge <- function(rnet_x, rnet_y, dist = 5, funs = NULL, sum_flows = TRUE,
   res_df = dplyr::bind_cols(res_list)
   res_sf = dplyr::left_join(rnet_x, res_df)
   if (sum_flows) {
+    print('-------------start----------')
     res_sf$length_x = as.numeric(sf::st_length(res_sf))
+      # Calculate the angle between 'corr_line_geometry' and 'geometry' for each row
+    res_sf$angle = sapply(1:nrow(res_sf), function(i) {
+      calculate_angle(get_vector(res_sf$corr_line_geometry[i,]), get_vector(res_sf$geometry[i,]))
+    })
+    print(names(res_sf))
     for(i in sum_cols) {
       # TODO: deduplicate
       length_y = as.numeric(sf::st_length(rnet_y))
+      mask <- (res_sf$angle < 15) | (res_sf$angle > 160)
```

Comment:
These are hard coded. At some point it would be good to make these available as arguments.

Reply:
> These are hard coded. At some point it would be good to make these available as arguments.

Got it, Robin. Sorry at the moment there is a bug in the code, I am working on it to solve it. (Switching from Python to R for programming is tough that I thought)

Reply:
Yes it's tricky. Keep up the good work and no worries about making these hardcoded values available as arguments, can add later..

Reply:
Hi @Robinlovelace got some results, catch up with you tomorrow
![image](https://github.com/ropensci/stplanr/assets/74598734/cd401387-996d-4668-85df-1ce3a9904581)